### PR TITLE
PP-6669 add payout date to settlement summary for transaction search

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT
- join payout column on gateway_payout_id to transactions
- surface to resource level paid_out_date as settled_date
- add relevant tests

### CODE REVIEW

File: src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java [hunk 1]
```diff
@@ -38,10 +38,15 @@
...
 
-    private static final String SEARCH_TRANSACTIONS = "SELECT * FROM transaction t " +
+    private static final String SEARCH_TRANSACTIONS = "SELECT t.*, po.paid_out_date AS paid_out_date FROM transaction t " +
+            "LEFT OUTER JOIN payout po on " +
+            "t.gateway_payout_id = po.gateway_payout_id " +
+            ":payoutAccountId " +
             ":searchExtraFields " +
             "ORDER BY t.created_date DESC OFFSET :offset LIMIT :limit";
 
+    private static final String SEARCH_TRANSACTIONS_WITH_PAYOUT = "AND po.gateway_account_id IN (<account_id>) ";
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java:48]: (Small one) Could it be renamed to indicate it is a clause or inline this as it is not full search query ?

File: src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java [hunk 2]
```diff
@@ -275,13 +280,17 @@ public Long getTotalWithLimitForSearch(TransactionSearchParams searchParams) {
...
             String baseQueryString
     ) {
-        String searchClauseTemplate = String.join(" AND ", filterTemplates);
+        String searchClauseTemplate = String.join(" AND ", searchParams.getFilterTemplates());
         searchClauseTemplate = StringUtils.isNotBlank(searchClauseTemplate) ?
                 "WHERE " + searchClauseTemplate :
                 "";
+        baseQueryString = baseQueryString
+                .replace(":payoutAccountId",
+                        (searchParams.getAccountIds() != null &&!searchParams.getAccountIds().isEmpty())
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java:292]:
```suggestion
                        (searchParams.getAccountIds() != null && !searchParams.getAccountIds().isEmpty())
```

File: src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java [hunk 3]
```diff
@@ -45,17 +47,28 @@ public TransactionEntity map(ResultSet rs, StatementContext ctx) throws SQLExcep
...
         return rs.wasNull() ? null : value;
     }
 
-    private Optional<ZonedDateTime> getZonedDateTime(ResultSet rs, String columnLabel) throws SQLException {
-        Timestamp timestamp = rs.getTimestamp(columnLabel);
+    private Optional<ZonedDateTime> getZonedDateTime(ResultSet rs, String columnLabel) {
+        Timestamp timestamp;
+        try {
+            timestamp = rs.getTimestamp(columnLabel);
+        } catch (SQLException ex) {
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java:69]: I think we can just throw SQLException here as we will need to join payout all the time (search transactions as well as get one transaction).

Comment (SandorArpa) [reply src/main/java/uk/gov/pay/ledger/transaction/dao/mapper/TransactionMapper.java:69]: As discussed, it's a temporary solution until all queries have the payout table joined to transaction table.

**Review by kbottla — APPROVED** [review]

**Pull request merged**